File: src/main/java/org/dedda/bratwurst/parse/PopParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.Pop;

import static org.dedda.bratwurst.parse.Emoji.*;
import static org.dedda.bratwurst.parse.Patterns.VARIABLE_NAME;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class PopParser extends InstructionParser {

    public Pop parse(String line, int linenumber) {
        if (line.matches(Patterns.POP)) {
            String name = line.substring(PINEAPPLE.length(), line.length() - MONKEY_FACE.length());
            if (!name.matches(VARIABLE_NAME)) {
                throw new RuntimeException("\"" + name + "\" is not a valid variable name!");
            }
            return new Pop(linenumber, name);
        }
        return null;
    }

}

File: src/main/java/org/dedda/bratwurst/parse/PopParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.Pop

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class PopParser : InstructionParser() {
    override fun parse(line: String, linenumber: Int): Pop? {
        if (line.matches(PATTERN)) {
            val name = line.substring(1, line.length - 1)
            return Pop(linenumber, name)
        }
        return null
    }

    companion object {
        private val PATTERN = Regex(Patterns.POP)
    }
}